src/main/scala/com/github/saurfang/sas/mapreduce/SasRecordReader.scala

```diff
@@ -43,32 +45,51 @@ class SasRecordReader(split: InputSplit,
 
   // Sanity-Check: Ensure file is not compressed.
   private val codec = new CompressionCodecFactory(jobConf).getCodec(filePath)
```

Comment:
Do we want to allow all possible compression codecs, or just some?
That is, should we check here if the codec is in an allowed list?

Reply:
This will need to be experimented with a splittable compression such as bzip2. These splittable compression algorithms may not create the InputSplit's that contain the full block to be parsed. So this may or may not support those (as the InputSplit might not contain the correct/whole sas block?) 

If that is the case then it should check a list of allowed.

Reply:
I was able to test and verify that externally splittable compressions such as bzip2 do not work with the sas reader, as it is not possible to seek to the previous page boundary in the sas file with a CompressionInputStream.

Setting isSplitable to false in SasInputFormat will make spark/hadoop always read externally compressed sas files in a single InputSplit, so any of the supported compression algorithms work, but they all have to be processed on a single executor.

Reply:
The splits for uncompressed data are just adjusted to fit the page boundaries, is this not possible with splitable compressed data? (Note the the "splits" are just offset numbers provided to each executor, and the executor can do whatever it likes)